pkg/controller/awsfederatedaccountaccess/awsfederatedaccountaccess_controller.go

```diff
@@ -157,6 +158,35 @@ func (r *ReconcileAWSFederatedAccountAccess) Reconcile(request reconcile.Request
 		return reconcile.Result{}, nil
 	}
 
+	// Check if the FAA has the uuid label
+	if !hasUUIDLabel(currentFAA) {
+		// Generate a new UUID
+		uuid := rand.String(6)
+
+		reqLogger.Info(fmt.Sprintf("Adding UUID %s to AccountAcess %s", uuid, currentFAA.Name))
+		newLabel := map[string]string{"uuid": uuid}
+
+		// Join the new UUID label with any current labels
+		if currentFAA.Labels != nil {
+			currentFAA.Labels = joinLabelMaps(currentFAA.Labels, newLabel)
+		} else {
+			currentFAA.Labels = newLabel
+		}
+
+		// Update the CR with new labels
+		err = r.client.Update(context.TODO(), currentFAA)
+		if err != nil {
+			reqLogger.Error(err, fmt.Sprintf("Lable update for %s failed", currentFAA.Name))
+			return reconcile.Result{}, err
+		}
+
+	}
+
+	uuidLabel, ok := currentFAA.Labels["uuid"]
```

Comment:
Could make this a function that returns a label and err since this same 4 lines is done a lot through code. Future improvement.

Reply:
+1